### PR TITLE
Add support for compass priority

### PR DESCRIPTION
**Pull request by almatthew**
[ACE-815](https://planckaero.atlassian.net/browse/ACE-815?atlOrigin=eyJpIjoiM2NmMWMwMTQyNGE5NDYwYTg4M2Y0ZDBhOTdkNTU0ZTEiLCJwIjoiaiJ9)

Addresses issue with APM v4.0.4 which removes the `COMPASS_PRIMARY` parameter.  This diff is a cherry-pick of https://github.com/mavlink/qgroundcontrol/commit/b1860d284ab709d7edfb4f6fc955f49a962c64fd#diff-bb218e3c78b2d638d2fa7ca26994c608e8b66049c00dd44c3fd113cc30f274cc.

### CODE REVIEW

File: src/FirmwarePlugin/APM/APMSensorParams.qml [hunk 1]
```diff
@@ -17,15 +17,23 @@ Item {
...
+    property bool compass2Primary:                  compassPrimaryFactAvailable ? compassPrimaryFact.rawValue == 1 : false
+    property bool compass3Primary:                  compassPrimaryFactAvailable ? compassPrimaryFact.rawValue == 2 : false
     property var  rgCompassPrimary:                 [ compass1Primary, compass2Primary, compass3Primary ]
 
     property Fact compass1Id:                       factPanelController.getParameterFact(-1, "COMPASS_DEV_ID")
     property Fact compass2Id:                       factPanelController.getParameterFact(-1, "COMPASS_DEV_ID2")
     property Fact compass3Id:                       factPanelController.getParameterFact(-1, "COMPASS_DEV_ID3")
+    property var  rgCompassId:                      [ compass1Id, compass2Id, compass3Id ]
+
+    property bool compassPrioFactsAvailable:        factPanelController.parameterExists(-1, "COMPASS_PRIO1_ID")
```
Comment (ptran) [inline src/FirmwarePlugin/APM/APMSensorParams.qml:32]: Does this really guarantee that `COMPASS_PRIO2_ID` and `COMPASS_PRIO3_ID` are available as well?

File: src/AutoPilotPlugins/APM/APMSensorsComponent.qml [hunk 2]
```diff
@@ -327,23 +341,54 @@ SetupPage {
...
+
+                            QGCComboBox {
+                                model:      [ qsTr("Priority 1"), qsTr("Priority 2"), qsTr("Priority 3"), qsTr("Not Set") ]
+                                visible:    _singleCompassSettingsComponentShowPriority && sensorParams.compassPrioFactsAvailable && useCompassCheckBox.visible && useCompassCheckBox.checked
+
+                                property int _compassIndex: index
+
+                                function selectPriorityfromParams() {
+                                    if (visible) {
+                                        currentIndex = 3
```
Comment (ptran) [inline src/AutoPilotPlugins/APM/APMSensorsComponent.qml:370]: I think this function always selects the last index item before cycling through and finding the right one. Would it make more sense to put this after the for loop? It's probably not noticeable in the UI.

**Review by ptran — COMMENTED** [review]
Left a couple comments, but overall, I think this looks good.

File: src/AutoPilotPlugins/APM/APMSensorsComponent.qml (hunk 2 above)
Comment (almatthew) [reply src/AutoPilotPlugins/APM/APMSensorsComponent.qml:370]: @ptran TBH this is just a cherry-pick from upstream QGC, so I don't have the answers to your questions. But, appears to work on the bench...

**Review by AubreyC — APPROVED** [review]

**Review by ptran — APPROVED** [review]

**Pull request merged**